python/cudf/cudf/_lib/cpp/scalar/scalar.pxd

```diff
@@ -51,3 +52,11 @@ cdef extern from "cudf/scalar/scalar.hpp" namespace "cudf" nogil:
         string_scalar(string st, bool is_valid) except +
         string_scalar(string_scalar other) except +
         string to_string() except +
+
+    cdef cppclass fixed_point_scalar[T](scalar):
+        fixed_point_scalar() except +
+        fixed_point_scalar(int64_t value,
+                           scale_type scale,
+                           bool is_valid) except +
+        int64_t value() except +
+        # TODO: Figure out how to add an int32 overload of value()
```

Comment:
Should we be using the template type `T` here instead of `int64_t`?

Reply:
The template type in practice ends up being `decimal64`, see `decimals.pxd` for the wrapper. But we don't actually want to return that - `value` returns an `int64_t` or an `int32_t` for `decimal64` and `decimal32` respectively. What we need in the long run is an overload of `value` that returns an `int32_t`, but we didn't really see an easy way of doing that in cython.

Reply:
The hacky workaround I think I've seen in the past here is to do something like `int32_t value2 "value"() except +`, maybe that would work?

Reply:
I think I see how that would work. I don't mind adding this, but only regret with it is that without plumbing at least some of the rest of the pieces of the `decimal32` pipeline, it might be kind of hard to tell if it works or not.

Reply:
Let's punt on it for now then, not a blocker for this PR.